service-worker.js

```diff
@@ -0,0 +1,21 @@
+// The cache identifier. Change this whenever the caching strategy changes.
+const cacheName = 'EIP-cache-v1';
+
+self.addEventListener('fetch', (event) => {
+  event.respondWith(async () => {
+    // Fetch cached response
+    let cache = await caches.open(cacheName);
+    let cachedResponse = await cache.match(event.request);
```

Comment:
Is this behaviour any different from how browsers normally work with the caching headers we already send?

Reply:
Yes. If the page has changed, then it will get updated when the page refreshes, instead of having to wait for the cache to expire.

Also, this cache has no expiration.